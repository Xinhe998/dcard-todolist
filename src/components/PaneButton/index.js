import React from 'react';
import PropTypes from 'prop-types';

const PaneButton = ({ icon, text }) => (
  <button className="PaneBtn" type="button">
    <span className="PaneBtn__icon">
      <img src={icon} draggable="false" alt="icon" />
    </span>
    <div className="PaneBtn__label">{text}</div>
  </button>
);
PaneButton.propTypes = {
  icon: PropTypes.string,
  text: PropTypes.string,
};
PaneButton.defaultProps = {
  icon: '',
  text: '',
};
export default PaneButton;
