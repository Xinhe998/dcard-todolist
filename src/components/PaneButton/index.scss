.PaneBtn {
  outline: 0;
  font-size: var(--xs-font-size);
  border-radius: 10px;
  background: #fafafa;
  border: 1px solid #d8d8d8;
  box-shadow: var(--panebtn-box-shadow);
  margin-top: 10px;
  transition: all 0.5s;
  display: inline-block;
  height: 70px;
  width: 120px;
  overflow: hidden;
  padding: 10px 7px 7px;
  cursor: pointer;
  margin-right: 20px;
  color: var(--text-color);
  &:hover {
    box-shadow: 0px 2px 10px 0 rgba(0, 0, 0, 0.3);
  }
  &:disabled {
    cursor: not-allowed;
    &:hover {
      box-shadow: var(--panebtn-box-shadow);
    }
    .PaneBtn__label {
      color: #d8d8d8;
    }
  }
  .PaneBtn__icon {
    display: block;
    img {
      width: 30px;
      height: 30px;
    }
  }
  .PaneBtn__label {
    display: block;
    line-height: 2;
  }
}