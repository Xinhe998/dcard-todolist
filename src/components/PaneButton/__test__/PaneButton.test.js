/* eslint-disable func-names */
/* eslint-disable no-unused-expressions */
/* eslint-disable object-curly-newline */
import React from 'react';
import PropTypes from 'prop-types';
import { shallow, configure, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PaneButton from '../index';

import calendar from '../../../assets/calendar.png';

// 以該解析器提供給 Enzyme 做渲染 Component 的設置
configure({ adapter: new Adapter() });

describe('<PaneButton />', function() {
  let testComp;
  beforeEach(() => {
    jest.resetModules();
    this.params = {
      icon: null,
      text: '',
      handleClick: jest.fn(),
      disabled: false,
    };

    this.makeSubject = () => {
      const { icon, text, handleClick, disabled } = this.params;

      testComp = shallow(
        <PaneButton
          icon={icon}
          text={text}
          handleClick={handleClick}
          disabled={disabled}
        />,
      );
      return testComp;
    };
  });

  describe('when it has complete data', () => {
    beforeEach(() => {
      this.params.icon = calendar;
      this.params.text = 'Test';
      this.subject = this.makeSubject();
    });

    it('should be same as snapshot', () => {
      const wrapper = render(testComp);
      expect(wrapper).toMatchSnapshot();
    });
    // 測試找到一個<button>
    it('should render one <button>', () => {
      expect(testComp.find('button')).toHaveLength(1);
    });
  });
});
