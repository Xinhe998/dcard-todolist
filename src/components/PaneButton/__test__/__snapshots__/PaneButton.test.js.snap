// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaneButton /> when it has complete data should be same as snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "PaneBtn",
      "type": "button",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "PaneBtn__icon",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "alt": "icon",
              "draggable": "false",
              "src": "calendar.png",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "alt": undefined,
              "draggable": undefined,
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "alt": undefined,
              "draggable": undefined,
              "src": undefined,
            },
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "PaneBtn__label",
          },
          "children": Array [
            Object {
              "data": "Test",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "PaneBtn__label",
        },
        "children": Array [
          Object {
            "data": "Test",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "PaneBtn__icon",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "alt": "icon",
                "draggable": "false",
                "src": "calendar.png",
              },
              "children": Array [],
              "name": "img",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "alt": undefined,
                "draggable": undefined,
                "src": undefined,
              },
              "x-attribsPrefix": Object {
                "alt": undefined,
                "draggable": undefined,
                "src": undefined,
              },
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "type": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "type": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
